### PR TITLE
Integrate dataset slicing into normal ascii data plugin

### PR DESCRIPTION
**Pull request by gassmoeller**
@jdannberg this is my suggestion for improving #4481. I like it more than duplicating plugins, because it is far less code (even though we still have to duplicate a bit inside the plugin). It should be reasonably simple to move this code up in the inheritance order (e.g. into AsciiDataInitial or even AsciiDataBase) if we want to make it available to other plugins. What do you think?

### CODE REVIEW

File: include/aspect/initial_composition/ascii_data.h [hunk 1]
```diff
@@ -79,6 +76,23 @@ namespace aspect
...
+        bool slice_data;
+
+        // The two points that define the location of the slice.
+        Tensor<1,3> first_point_on_slice;
+        Tensor<1,3> second_point_on_slice;
+
+        /**
+          * The matrix that describes the rotation by which a 2D model
+          * needs to be transformed to a plane that contains the origin and
+          * the prescribed point given in the input.
```
Comment (jdannberg) [inline include/aspect/initial_composition/ascii_data.h:93]: This needs to be updated. 
```suggestion
          * the two prescribed points given in the input.
```

File: source/initial_composition/ascii_data.cc [hunk 2]
```diff
@@ -59,6 +76,31 @@ namespace aspect
...
                                                           "box_2d.txt");
+
+        prm.enter_subsection("Ascii data model");
+        {
+          prm.declare_entry("Slice dataset in 2D plane", "false",
+                            Patterns::Bool (),
+                            "Whether to use a 2D data slice from a 3D data file "
+                            "or the entire data file. Slicing a 3D dataset is "
+                            "only supported for 2D models.");
+          prm.declare_entry ("First point on slice", "0.0,1.0,0.0",
```
Comment (jdannberg) [inline source/initial_composition/ascii_data.cc:87]: Can you add a comment that these parameters will be ignored if "Slice dataset in 2D plane" is false.

File: tests/ascii_data_slice.prm [hunk 3]
```diff
@@ -50,14 +50,12 @@ end
...
-  set List of model names = ascii data slice
+  set List of model names = ascii data
 
   subsection Ascii data model
     set Data directory = $ASPECT_SOURCE_DIR/data/initial-composition/ascii-data/test/
     set Data file name = shell_3d_2.txt
-  end
-
-  subsection Ascii data slice
+    set Slice data in 2D plane = true
```
Comment (jdannberg) [inline tests/ascii_data_slice.prm:58]:
```suggestion
    set Slice dataset in 2D plane = true
```
This is why the test is failing.

**Review by jdannberg — APPROVED** [review]
I like this much better than my version! Thank you!

What do you think about the coordinate system? Should we add that as an option in a follow-up PR (especially if we want to make versions of this for other ascii data plugins)?

**Comment by gassmoeller** [comment]
Thanks for the comments, I think I have addressed them in the last commit. 

> This is why the test is failing.

Yes, I was experimenting with different parameter names. Do you think the current one is ok?

**Comment by jdannberg** [comment]
Looks good, thank you!

For the name: I was confused at first, since 'slice' can also be used as a noun, so I was wondering what's a 'slice dataset'? But I don't have a better idea, so I think this is fine.

**Pull request merged**